frontend-react/src/App.tsx

```diff
@@ -64,6 +64,12 @@ const App = () => {
         navigate(toRelativeUrl(originalUri, window.location.origin));
     };
 
+    // watch the session broadcast channel to handle login/logout on other tabs
+    const bc = new BroadcastChannel("session");
+    bc.onmessage = function () {
```

Comment:
Since BroadcastChannel can be used for any event, let's narrow it down by checking the name:

```jsx
bc.onmessage = (event) => {
  // could also do something with event.data, which is 'login' or 'logout'
  if (event.currentTarget.name === 'session') {
     OKTA_AUTH.authStateManager.updateAuthState();
  }
};

// Side note: I think we could also use attachEventListener, 
// which is more modern since on<event> nomenclature was from the olden days of HTML event-binding:
bc.addEventListener('message', (event) => {
  if (event.currentTarget.name === 'session') {
     OKTA_AUTH.authStateManager.updateAuthState();
  }
});
```

ALSO, what do you think about enumifying the 'session', 'login', and 'logout'?  At least to prevent typos going forward :)

Reply:
Great point on enumifying, do you have a good suggestion for where that enum should live? Perhaps `SessionStorageTools` or `UserUtils` ?

Should not be any need to check for `session`. The BroadcastChannel is named on instantiation (`new BroadcastChannel("session")`) so these event listeners will only ever receive `session` messages. 

I totally agree we might want to check for `logout` vs `login` (as mentioned in Notes), but for now those are the only two actions and they are both handled the same, so I vote we add that check later on.

Reply:
SessionStorageTools makes sense to me since that's likely the only place we'll be using this for now!

The idea of checking for `session` was because we're creating two `session` instances of BroadcastChannel [here](https://github.com/CDCgov/prime-reportstream/pull/7727/files#diff-f69b57e78f568071b4f3cab66ca28cc47c5efae1b0959c5f4eef858538b1ac02R18) and [here](https://github.com/CDCgov/prime-reportstream/pull/7727/files#diff-64a2066f31389cb7d53d6360524cfcb49c7fecd153a2db15db0aa49b92f954efR7) -- if we update the name for whatever reason and forget to change it in one place, that'll fail to respond properly to the events we're sending.  HOWEVER, this is a moot point if we only choose [to create one instance](https://github.com/CDCgov/prime-reportstream/pull/7727/files#r1053669075) like we're discussing below!